### PR TITLE
Add support for richdocuments

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
- open editable documents (text, spreadsheet, presentation) directly in webview 
- support sharing within view
- insert local file in document
- insert remote file in document

TODO
- [x] @rullzer add brigde call once UI is completely loaded

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/main/java/com/owncloud/android/datamodel/Template.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @ParcelConstructor
+    public Template(int id, String name, String thumbnailLink, String type, String extension) {
+        this.id = id;
+        this.name = name;
+        this.thumbnailLink = thumbnailLink;
+        this.type = type;
+        this.extension = extension;
+    }
+
+    public int getId() {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/datamodel/Template.java]: shouldn't we make the getter/setter disappear and make use of lombok?

**Comment by AndyScherzinger** [comment]
👍 Just some minor comments regarding use of lombok, nullcheck, view id

Can't test since I don't have collabora (to check the UI parts)

@tobiasKaminsky could you popst some screenshots or mail me a server/user where I could test this during the next days?

**Comment by AndyScherzinger** [comment]
Might need a rebase - finbugs count is up by 4 warnings compared to master.

**Comment by tobiasKaminsky** [comment]
Findbug should be no problem, as it will get updated when merging, but feel free to rebase after your test and then merge :+1:

**Pull request merged**